lightning/src/routing/router.rs

```diff
@@ -1448,16 +1460,21 @@ where L::Target: Logger {
 	drawn_routes.sort_by_key(|paths| paths.iter().map(|path| path.get_total_fee_paid_msat()).sum::<u64>());
 	let mut selected_paths = Vec::<Vec<Result<RouteHop, LightningError>>>::new();
 	for payment_path in drawn_routes.first().unwrap() {
-		selected_paths.push(payment_path.hops.iter().map(|(payment_hop, node_features)| {
+		let mut path = payment_path.hops.iter().map(|(payment_hop, node_features)| {
 			Ok(RouteHop {
 				pubkey: PublicKey::from_slice(payment_hop.node_id.as_slice()).map_err(|_| LightningError{err: format!("Public key {:?} is invalid", &payment_hop.node_id), action: ErrorAction::IgnoreAndLog(Level::Trace)})?,
 				node_features: node_features.clone(),
-				short_channel_id: payment_hop.short_channel_id,
-				channel_features: payment_hop.channel_features.clone(),
+				short_channel_id: payment_hop.candidate.short_channel_id(),
+				channel_features: payment_hop.candidate.features(),
 				fee_msat: payment_hop.fee_msat,
-				cltv_expiry_delta: payment_hop.cltv_expiry_delta,
+				cltv_expiry_delta: payment_hop.candidate.cltv_expiry_delta(),
 			})
-		}).collect());
+		}).collect::<Vec<_>>();
+		let mut prev_cltv_expiry_delta = final_cltv_expiry_delta;
+		for hop in path.iter_mut().rev() {
+			core::mem::swap(&mut prev_cltv_expiry_delta, &mut hop.as_mut().unwrap().cltv_expiry_delta);
+		}
```

Comment:
I see tests fail without this but I'm confused on what it's accomplishing, comment?

Reply:
Right, this was previously in the `if node_id == our_node_id` loop, above, see comment there for more info, it moved here to handle moving the data into the `CandidateRouteHop`.